src/notte/controller/base.py

```diff
@@ -94,18 +99,43 @@ async def execute_interaction_action(self, action: InteractionAction) -> Browser
         if action.press_enter is not None:
             press_enter = action.press_enter
         # locate element (possibly in iframe)
-        locator = await locale_element(self.window.page, action.selector)
+        locator: Locator = await locale_element(self.window.page, action.selector)
         original_url = self.window.page.url
 
         action_timeout = self.window.config.wait.action_timeout
 
+        def is_code(text: str) -> bool:
+            try:
+                lexer = guess_lexer(text)
+                return str(lexer) != "<pygments.lexers.TextLexer>"
+            except Exception:
+                return False
+
         match action:
             # Interaction actions
             case ClickAction():
                 await locator.click(timeout=action_timeout)
             case FillAction(value=value):
-                await locator.fill(value, timeout=action_timeout)
-                await self.window.page.wait_for_timeout(500)
+                if is_code(text=value):
```

Comment:
can add a logging message: "Code detected in fill action: simulate clipboard copy/paste for better string formating" or smeting like that